### ISSUE TITLE
Searching within book lists

### ISSUE THREAD

**Issue by el4ctr0n** [internetarchive/openlibrary#10220]
### Proposal

Dear developers,
Can you please make it possible to search within a book list (similar to how it's already possible on author pages)?

Thank you for considering it.

#### Requirements Checklist

* [ ]

#### Related files

*

#### Stakeholders

*

<hr>

#### Instructions for Contributors

Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

**Comment by ananyakaligal** [comment]
@el4ctr0n could i work on this issue?

**Comment by ananyakaligal** [comment]
@el4ctr0n  Yes, this is a great feature to add! It would make navigating specific book lists much easier and enhance the overall user experience. If the maintainers approve this, I would be happy to work on adding this feature. @mekarpeles @RayBB

**Comment by mekarpeles** [comment]
This one is a nice idea! Searching lists is a challenge which likely requires one of two solution:

1. fetching all books IDs from a list (which could have thousands of books) and then passing it in with the query to the search.json API
2. adding lists into solr directly, which may be tough to keep up to date and may also have privacy consequences.

@benbdeitch @scottbarnes and @cdrini are important stakeholders here. I personally don't know that there's enough usage of [large] lists to justify this feature at this time, though that might depend on how much work is required and if a community member can easily re-use:
* #7052 which addressed #5080

**Comment by el4ctr0n** [comment]
A major use case for this feature would be the various search operators, for example, ia:* to hide undigitized books from a book list.
